File: packages/create-bookmarklet/templates/python/src/features/hello.py

from browser import document, html, alert

def create_interface():
    container = html.DIV(style={
        'background': 'white',
        'padding': '20px',
        'border-radius': '8px',
        'box-shadow': '0 2px 10px rgba(0,0,0,0.1)'
    })

    title = html.H1("Python Bookmarklet", style={'margin': '0 0 15px 0'})
    button = html.BUTTON("Click Me", Class="btn-primary")

    def handle_click(ev):
        alert("Hello from Python!")

    button.bind('click', handle_click)

    container <= title
    container <= button

    return container

File: packages/create-bookmarklet/templates/javascript/src/features/hello.js

export function createHello() {
    const container = document.createElement('div');
    container.style.cssText = `
        background: white;
        padding: 20px;
        border-radius: 8px;
        box-shadow: 0 2px 10px rgba(0,0,0,0.1);
        font-family: system-ui, sans-serif;
    `;

    const title = document.createElement('h1');
    title.textContent = 'JavaScript Bookmarklet';
    title.style.margin = '0 0 15px 0';

    const button = document.createElement('button');
    button.textContent = 'Click Me';
    button.style.cssText = `
        padding: 8px 16px;
        background: #007bff;
        color: white;
        border: none;
        border-radius: 4px;
        cursor: pointer;
    `;

    button.addEventListener('click', () => {
        alert('Hello from JavaScript!');
    });

    container.appendChild(title);
    container.appendChild(button);
    return container;
}
